### PR TITLE
Add files-watcher for 'start' command

### PR DESCRIPTION
**Pull request by nicoecheza**

### CODE REVIEW

File: packages/@dcl/sdk/cli/commands/preview/wire.ts [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
 import { setupEcs6Endpoints } from './endpoints'
 import { CliError } from '../../utils/error'
 import { PreviewComponents } from './types'
+import { upgradeWebSocketResponse } from '@well-known-components/http-server/dist/ws'
 
-export async function wire(dir: string, components: PreviewComponents) {
+export async function wire(
+  components: Pick<CliComponents, 'fs'>,
+  dir: string,
+  previewComponents: PreviewComponents,
```
Comment (menduz) [inline packages/@dcl/sdk/cli/commands/preview/wire.ts:20]:
```suggestion
  components: CliComponents,
  dir: string,
```

Please unify components into CliComponents, delete PreviewComponents

**Review by menduz — CHANGES_REQUESTED** [review]

File: packages/@dcl/sdk/cli/commands/preview/wire.ts [hunk 2]
```diff
@@ -17,10 +29,73 @@ export async function wire(dir: string, components: PreviewComponents) {
...
+
+  if (watch) {
+    const { clients } = previewComponents.ws.ws
+    chokidar
+      .watch(resolve(dir, 'bin'), {
+        ignoreInitial: false,
+        cwd: dir
+      })
+      .on('all', async (_, file) => {
+        if (await shouldUpdateScene(components, dir, file)) {
```
[the remark this answers is not in the document]
Comment (nicoecheza) [reply packages/@dcl/sdk/cli/commands/preview/wire.ts:57]: we are only watching files inside the `bin` directory :D

[the remark this answers is not in the document]
Comment (nicoecheza) [reply packages/@dcl/sdk/cli/commands/preview/wire.ts:55]: fixed 👌

**Review by menduz — APPROVED** [review]

File: packages/@dcl/sdk/cli/commands/preview/wire.ts [hunk 3]
```diff
@@ -17,10 +24,84 @@ export async function wire(dir: string, components: PreviewComponents) {
...
+  dir: string,
+  file: string
+): Promise<boolean> => {
+  const sceneFile = await getSceneFile(components, dir)
+  if (resolve(file) !== resolve(dir, sceneFile.main)) {
+    // ignore source files
+    if (file.endsWith('.ts') || file.endsWith('.tsx')) {
+      return false
+    }
+  }
```
[the remark this answers is not in the document]
Comment (nicoecheza) [reply packages/@dcl/sdk/cli/commands/preview/wire.ts:90]: this condition is useless if you have a proper `.dclignore`, but leaving it there won't harm anyone for now. We should add those extensions + node_modules + some-other-stuff to the defaults to make sure we don't add watchers for unnecessary files. 

Issue for this created [here](https://github.com/decentraland/sdk/issues/599)

**Pull request merged**